### ISSUE TITLE
Add support for MATLAB codecheck, codeIssues

### ISSUE THREAD

**Issue by stv0g** [sco1/pre-commit-matlab#42]
There are several code-quality tools provided by MATLAB:
- [mlint](https://de.mathworks.com/help/matlab/ref/mlint.html) (is deprecated in favor of checkcode)
- [checkcode](https://de.mathworks.com/help/matlab/ref/checkcode.html) (is deprecated in favor of codeIssues)
- [codeIssues](https://de.mathworks.com/help/matlab/ref/codeissues.html)

It would be great if we could add support for running the MATLAB code quality checkers in a dedicated pre-commit hook.

**Comment by sco1** [comment]
As far as I can tell the latter two are MATLAB only, the [MATLAB Language Server](https://github.com/mathworks/MATLAB-language-server) may facilitate these but I am not familiar with its use so not sure if it would help here.

mlint may be pretty feasible since it has a standalone executable. I'll try and take a look at it this week and see what can be accomplished.

**Comment by stv0g** [comment]
> As far as I can tell the latter two are MATLAB only

Yes that's true. But mlint itself is a MATLAB function as well. I think it should be pretty straight forward to wrap the MATLAB function into a standalone script/executable to work around this.

The language server sounds quite interesting too, but I am looking for running the checks in our CI too. Maybe there is a pre-commit hook to single-run the code through a LSP language server? That would be interesting for other languages as well.

There is also the pretty nice VSCode extension by Mathworks themselves: https://github.com/mathworks/MATLAB-extension-for-vscode

But I don't think it's using the language server?

**Comment by sco1** [comment]
> I think it should be pretty straight forward to wrap the MATLAB function into a standalone script/executable to work around this.

I do not have a personal license for the MATLAB Compiler, and even if I did the MATLAB Runtime is version-specific and ensuring that either myself or the end users or both are synchronized isn't a maintenance burden I want to handle. Keeping track of spawning MATLAB instances for every commit is also out of scope for me, and Windows still lacks a headless launch option.

> The language server sounds quite interesting too, but I am looking for running the checks in our CI too. Maybe there is a pre-commit hook to single-run the code through a LSP language server?

The MathWorks maintains a [set of GitHub actions](https://github.com/matlab-actions) that may be more useful for use-cases that require managing an existing MATLAB installation.

The ideal path would likely be to add MATLAB language support to pre-commit (see: [adding support for a new hook language](https://github.com/pre-commit/pre-commit/blob/main/CONTRIBUTING.md#adding-support-for-a-new-hook-language), likely through the use of the language server. I have had this thought on occasion but has not been an effort I have been able to take on.

> But I don't think it's using the language server?

It does. It is [vendored as a submodule](https://github.com/mathworks/MATLAB-extension-for-vscode/blob/main/.gitmodules).
